packages/ui/src/modal/buy/BuyModal.tsx

```diff
@@ -226,21 +244,50 @@ export function BuyModal({
         const lastStepItems =
           executableSteps[executableSteps.length - 1]?.items || []
 
-        const totalPurchases =
-          stepData?.currentStep?.items?.reduce((total, item) => {
-            item.transfersData?.forEach((transferData) => {
-              total += Number(transferData.amount || 1)
-            })
-            return total
-          }, 0) || 0
+        const totalPurchases = creditCardCheckoutStatus
+          ? 1
+          : stepData?.currentStep?.items?.reduce((total, item) => {
+              item.transfersData?.forEach((transferData) => {
+                total += Number(transferData.amount || 1)
+              })
+              return total
+            }, 0) || 0
 
-        const failedPurchases = quantity - totalPurchases
-        const successfulPurchases = quantity - failedPurchases
+        const failedPurchases =
+          creditCardCheckoutStatus === 'PROCESSING_ERROR'
+            ? 1
+            : quantity - totalPurchases
+        const successfulPurchases =
+          creditCardCheckoutStatus === 'TRANSFER_SUCCEEDED'
+            ? 1
```

Comment:
Does paper allow for partial purchases if buying multiple 1155s? or is it all or nothing?